[gamemodes]/[play]/play/broph.lua

```diff
@@ -1,176 +1,132 @@
+local vehsToSpawn = {
+	-- {model id, x, y, z, r}
+	{411, -2507.703125, -334.5881652832, 58.109756469727, 140}
 }
 
-local spawnpoints =
-{
-	{-711,957,12.4,90},
-	{2005,1543,13.5,270},
-	{2485,-1667,13.3,0},
-	{-2405,-598,132.6,128},
+local spawnPosition = {
+	-- {x, y, w, r}
+	{-2511.6953125, -342.90386962891, 58.481922149658, 90},
 }
 
-local vehicleDestroyTimers = {}
-local validSkins = {0, 1, 2, 7, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 66, 67, 68, 69, 70, 71, 72, 73, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312}
-local playerVehicles = {}
+local defaultTimes = {
+	["VehRespawn"] = 60000,
+	["Expire"] = 600000,
+	["PlayerRespawn"] = 4000
+}
 
-local function spawn(player)
+local vehTimeToDestroy = {}
+local playerVeh = {}
 
-	if player and isElement(player) then
-		local x,y,z,r = unpack(spawnpoints[math.random(1,#spawnpoints)])
-		spawnPlayer(player,x+math.random(-3,3),y+math.random(-3,3),z,r,validSkins[math.random(1,#validSkins)],0,0)
-		fadeCamera(player, true)
-		setCameraTarget(player, player)
+addEventHandler("onResourceStop", resourceRoot, function()
+	destroyElements()
+end)
+
+addEventHandler("onResourceStart", resourceRoot, function()
+	for _, players in ipairs(getElementsByType("player")) do
+		spawn(players)
 	end
-end
 
-local function onJoin()
+	for index, value in ipairs(vehsToSpawn) do
+		createVeh(value)
+	end
+end)
 
+addEventHandler("onPlayerJoin", root, function()
 	spawn(source)
+end)
 
-end
+addEventHandler("onPlayerQuit", root, function()
+	destroyElements()
+end)
 
-local function onWasted()
+addEventHandler("onPlayerWasted", root, function()
+	local playerRespawn = defaultTimes["PlayerRespawn"]
+	setTimer(spawn, (playerRespawn > 50 and playerRespawn or 50), 1, source)

```

Comment:
Another tip: in these cases, you can simply use math.min or max.max to do the trick...
```suggestion
	setTimer(spawn, math.max(playerRespawn, 50), 1, source)
```
http://lua-users.org/wiki/MathLibraryTutorial